keras_nlp/models/backbone.py

```diff
@@ -28,6 +28,16 @@ def __init__(self, *args, dtype=None, **kwargs):
             id(layer) for layer in self._flatten_layers()
         )
         self._initialized = True
+        if dtype is not None:
```

Comment:
Thanks for this fix, I have one comment about setting the dtype. if you would like to give it a look, I will appreciate it.
I think we should follow how `keras.layer` sets the dtype police.
something like this:
```python
if config.keras_3():
    self.dtype_policy = keras.src.dtype_policies.get(dtype)
else:
    self._set_dtype_policy(dtype)
```
I think that will be better user experience as whatever `dtype` works for layer should be expected to work for the backbone, and for keras3 for example, `keras.src.dtype_policies.get(dtype)` covers more scenarios. 
also keras2 do some extra stuff after setting the dtype policy, it sets an attribute named `_compute_dtype_object`, I don't know its usage specifically, but it's used by a lot of functions in the class.